android/src/main/java/com/swmansion/reanimated/layoutReanimation/AnimationsManager.java

```diff
@@ -512,16 +356,151 @@ public void updateLayout(
+    int tag = view.getId();
+    boolean hasExitAnimation = hasAnimationForTag(tag, "exiting");
+    boolean wantAnimateExit = hasExitAnimation;
+
+    // we might want to keep this view around
+    // because one of the (children's) children
+    // has an exiting animation
+    if (view instanceof ViewGroup) {
+      ViewGroup viewGroup = (ViewGroup) view;
+      for (int i = viewGroup.getChildCount() - 1; i >= 0; i--) {
+        View child = viewGroup.getChildAt(i);
+        if (removeOrAnimateExitRecursive(child, viewGroup, shouldRemove && !hasExitAnimation)) {
+          wantAnimateExit = true;
+        }
+      }
+    }
+
+    if (hasExitAnimation) {
+      Snapshot before = new Snapshot(view, mReanimatedNativeHierarchyManager);
+      HashMap<String, Object> currentValues = before.toCurrentMap();
+      HashMap<String, Float> preparedValues = prepareDataForAnimationWorklet(currentValues, false);
+      if (!mExitingViews.containsKey(tag)) {
+        mExitingViews.put(tag, view);
+        registerExitingAncestors(view);
+        mNativeMethodsHolder.startAnimation(tag, "exiting", preparedValues);
+        mNativeMethodsHolder.clearAnimationConfig(tag);
+      }
+      return true;
+    }
+
+    mNativeMethodsHolder.clearAnimationConfig(tag);
+
+    if (wantAnimateExit) {
+      mAncestorsToRemove.add(tag);
+      return true;
+    }
+
+    if (shouldRemove) {
+      removeView(view, parent);
+    } else {
+      view.setClickable(false);
+    }
+    return false;
+  }
+
+  private void registerExitingAncestors(View view) {
+    View parent = (View) view.getParent();
+    while (parent != null && !(parent instanceof RootView)) {
+      int tag = parent.getId();
+      Integer previousValue = mExitingSubviewCountMap.get(tag);
+      int newValue = previousValue != null ? previousValue + 1 : 1;
+      mExitingSubviewCountMap.put(tag, newValue);
+
+      parent = (View) parent.getParent();
+    }
+  }
+
+  private void maybeDropAncestors(View exitingView) {
+    View parent = (View) exitingView.getParent();
+    while (parent != null && !(parent instanceof RootView)) {
+      View view = parent;
+      parent = (View) view.getParent();
+      int tag = view.getId();
+      Integer ancestorOfCount = mExitingSubviewCountMap.get(tag);
+      ancestorOfCount = ancestorOfCount != null ? ancestorOfCount - 1 : 0;
+      if (ancestorOfCount <= 0) {
+        if (mAncestorsToRemove.contains(tag)) {
+          mAncestorsToRemove.remove(tag);
+          if (!mExitingViews.containsKey(tag)) {
+            removeView(view, (ViewGroup) parent);
+          }
+        }
+        mExitingSubviewCountMap.remove(tag);
+      } else {
+        mExitingSubviewCountMap.put(tag, ancestorOfCount);
+      }
+    }
+  }
+
+  private void removeView(View view, ViewGroup parent) {
+    int tag = view.getId();
+    if (mCallbacks.containsKey(tag)) {
+      Runnable callback = mCallbacks.get(tag);
+      mCallbacks.remove(tag);
+      if (callback != null) {
```

Comment:
What we should to do if callback is null?

Reply:
Not run it? 🤷 
We don't insert nulls into the map, and we check the map contains key, so it shouldn't actually happen.